src/audio/module_adapter/module_adapter.c

```diff
@@ -963,7 +963,27 @@ int module_get_large_config(struct comp_dev *dev, uint32_t param_id, bool first_
 						  (uint8_t *)data, fragment_size);
 	return 0;
 }
+
+int module_adapter_get_attribute(struct comp_dev *dev, uint32_t type, void *value)
+{
+	struct processing_module *mod = comp_get_drvdata(dev);
+
+	switch (type) {
+	case COMP_ATTR_BASE_CONFIG:
+		memcpy_s(value, sizeof(struct ipc4_base_module_cfg), mod->priv.private,
+			 sizeof(struct ipc4_base_module_cfg));
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
 #else
+int module_adapter_get_attribute(struct comp_dev *dev, uint32_t type, void *value)
+{
+	return 0;
```

Comment:
this is IPC3, it will never be invoked.

Reply:
ok, lets have an inline comment so everyone follows.

Reply:
get_attribute is a generic function also used in ipc3. Anyone can try to take any attribute from any component. It would be nice to return an error if the function doesn't support the given attribute. Otherwise, the caller may mistakenly assume that the correct value was placed at the given address.